flying-saucer-core/src/main/java/org/xhtmlrenderer/layout/SharedContext.java

```diff
@@ -318,16 +316,16 @@ public NamespaceHandler getNamespaceHandler() {
 
     public void addBoxId(String id, Box box) {
         if (idMap == null) {
-            idMap = new HashMap();
+            idMap = new HashMap<>();
```

Comment:
Would it be more thread-safe to initialize `idMap` upon construction? Then there's no race-condition between the check for `null` and the `put()` call. If the map needs to be set to `null`, it can be `clear()`ed instead?

This would apply to lines 325-328 as well.

Reply:
Yes, it's a good point. 
Most of these changes, I did just by applying IDEA shortcuts etc. It's quite hard to review & improve all of them with a single PR. My philosophy is "at least do not make it worser". :)

Reply:
done